src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.Node.cs

```diff
@@ -72,5 +78,87 @@ private string GetDebuggerDisplay()
                 return NameSpan + ", " + ParentIndex;
             }
         }
+
+        public readonly struct ParameterTypeInfo
+        {
+            public string Name { get; }
+            public bool IsComplexType { get; }
+
+            public ParameterTypeInfo(string name, bool isComplex)
+            {
+                Name = name;
+                IsComplexType = isComplex;
+            }
+
+            public override string ToString()
+                => $"{Name}, {IsComplexType}";
+        }
+
+        public readonly struct ExtensionMethodInfo
+        {
+            public string FullyQualifiedContainerName { get; }
+
+            public string Name { get; }
+
+            public ExtensionMethodInfo(string fullyQualifiedContainerName, string name)
+            {
+                FullyQualifiedContainerName = fullyQualifiedContainerName;
+                Name = name;
+            }
+        }
+
+        private sealed class ParameterTypeInfoProvider : ISignatureTypeProvider<ParameterTypeInfo, object>
+        {
+            public static readonly ParameterTypeInfoProvider Instance = new ParameterTypeInfoProvider();
+
+            private static ParameterTypeInfo ComplexInfo
+                => new ParameterTypeInfo(string.Empty, isComplex: true);
+
+            public ParameterTypeInfo GetPrimitiveType(PrimitiveTypeCode typeCode)
+                => new ParameterTypeInfo(typeCode.ToString(), isComplex: false);
```

Comment:
who defines PrimitiveTypeCode?  And how do we know that typecode.ToString is correct for this?

Reply:
open question.

Reply:
open question :)

Reply:
It's declared in SRM (copied below), I couldn't find helper that does this, but it seems ToString() would do the right thing for our purpose.

```cs
namespace System.Reflection.Metadata
{
    //
    // Summary:
    //     Specifies constants that define primitive types found in metadata signatures.
    public enum PrimitiveTypeCode : byte
    {
        //
        // Summary:
        //     A System.Void type.
        Void = 1,
        //
        // Summary:
        //     A System.Boolean type.
        Boolean = 2,
        //
        // Summary:
        //     A System.Char type.
        Char = 3,
        //
        // Summary:
        //     An System.SByte type.
        SByte = 4,
        //
        // Summary:
        //     A System.Byte type.
        Byte = 5,
        //
        // Summary:
        //     A System.Int16 type.
        Int16 = 6,
        //
        // Summary:
        //     A System.UInt16 type.
        UInt16 = 7,
        //
        // Summary:
        //     A System.Int32 type.
        Int32 = 8,
        //
        // Summary:
        //     A System.UInt32 type.
        UInt32 = 9,
        //
        // Summary:
        //     A System.Int64 type.
        Int64 = 10,
        //
        // Summary:
        //     A System.UInt64 type.
        UInt64 = 11,
        //
        // Summary:
        //     A System.Single type.
        Single = 12,
        //
        // Summary:
        //     A System.Double type.
        Double = 13,
        //
        // Summary:
        //     An System.Single type.
        String = 14,
        //
        // Summary:
        //     A typed reference.
        TypedReference = 22,
        //
        // Summary:
        //     A System.IntPtr type.
        IntPtr = 24,
        //
        // Summary:
        //     A System.UIntPtr type.
        UIntPtr = 25,
        //
        // Summary:
        //     An System.Object type.
        Object = 28
    }
}
```

Reply:
#resolved